packages/lab/src/date-picker/DatePicker.tsx

```diff
@@ -0,0 +1,167 @@
+      controlled: startDateProp,
+      default: defaultStartDate,
+      name: "StartDate",
+      state: "value",
+    });
+    const [endDate, setEndDate] = useControlled({
+      controlled: endDateProp,
+      default: defaultEndDate,
+      name: "EndDateInput",
+      state: "value",
+    });
+
+    const { x, y, strategy, elements, floating, reference, context } =
+      useFloatingUI({
+        open: open,
+        onOpenChange: setOpen,
+        placement: "bottom-start",
+        middleware: [
+          size({
+            apply({ rects, elements, availableHeight }) {
+              Object.assign(elements.floating.style, {
+                minWidth: `${rects.reference.width}px`,
+                maxHeight: `max(calc((var(--salt-size-base) + var(--salt-spacing-100)) * 5), calc(${availableHeight}px - var(--salt-spacing-100)))`,
+              });
+            },
+          }),
+          flip({ fallbackStrategy: "initialPlacement" }),
+        ],
+      });
+
+    const dismiss = useDismiss(context);
+
+    const { getReferenceProps, getFloatingProps } = useInteractions([dismiss]);
+    const { disabled: formFieldDisabled, readOnly: formFieldReadOnly } =
+      useFormFieldProps();
+
+    const getPanelPosition = () => ({
+      top: y ?? 0,
+      left: x ?? 0,
+      position: strategy,
+      width: elements.floating?.offsetWidth,
+      height: elements.floating?.offsetHeight,
+    });
+
+    const panelRef = useRef<HTMLDivElement>(null);
+    const startInputRef = useRef<HTMLInputElement>(null);
+    const endInputRef = useRef<HTMLInputElement>(null);
+    const inputRef = useForkRef<HTMLDivElement>(ref, reference);
+    const floatingRef = useForkRef<HTMLDivElement>(panelRef, floating);
+
+    // Handlers
+    const handleSelect = () => {
+      selectionVariant === "range" && startDate
+        ? endInputRef?.current?.focus()
+        : startInputRef?.current?.focus();
+    };
+    const handleCalendarButton = () => {
+      startInputRef?.current?.focus();
+      setOpen(!open);
+    };
+
+    // Context
+    const datePickerContextValue = {
+      openState: open,
+      setOpen,
+      disabled,
+      endDate,
+      setEndDate,
+      defaultEndDate,
+      startDate,
+      setStartDate,
+      defaultStartDate,
+      selectionVariant,
+      context,
+      getPanelPosition,
+    };
+
+    return (
+      <DatePickerContext.Provider value={datePickerContextValue}>
+        <DateInput
+          className={clsx(withBaseName())}
+          ref={inputRef}
+          {...getReferenceProps()}
+          startInputRef={startInputRef}
+          endInputRef={endInputRef}
+          placeholder={placeholder}
+          dateFormatter={dateFormatter}
+          endAdornment={
+            <Button
+              variant="secondary"
+              onClick={handleCalendarButton}
+              disabled={disabled || formFieldReadOnly || formFieldDisabled}
+            >
+              <CalendarIcon />
```

Comment:
```suggestion
              <CalendarIcon aria-hidden />
```
We need to follow Icon button guidance